app/models/directory_item.rb

```diff
@@ -64,68 +56,38 @@ def self.refresh_period!(period_type, force: false)
       )
 
+      # Create user records for newer users who don't yet have a directory item
       add_missing_users(period_type)
+      query_args = period_query_args(period_type)
 
       # Calculate new values and update records
       #
-      #
       # TODO
       # WARNING: post_count is a wrong name, it should be reply_count (excluding topic post)
-      #
-      #
-      query_args = {
-        period_type: period_types[period_type],
-        since: since,
-        like_type: UserAction::LIKE,
-        was_liked_type: UserAction::WAS_LIKED,
-        new_topic_type: UserAction::NEW_TOPIC,
-        reply_type: UserAction::REPLY,
-        regular_post_type: Post.types[:regular],
-      }
-
-      DB.exec(
-        "WITH x AS (SELECT
-                    u.id user_id,
-                    SUM(CASE WHEN p.id IS NOT NULL AND t.id IS NOT NULL AND ua.action_type = :was_liked_type THEN 1 ELSE 0 END) likes_received,
-                    SUM(CASE WHEN p.id IS NOT NULL AND t.id IS NOT NULL AND ua.action_type = :like_type THEN 1 ELSE 0 END) likes_given,
-                    COALESCE((SELECT COUNT(topic_id) FROM topic_views AS v WHERE v.user_id = u.id AND v.viewed_at > :since), 0) topics_entered,
-                    COALESCE((SELECT COUNT(id) FROM user_visits AS uv WHERE uv.user_id = u.id AND uv.visited_at > :since), 0) days_visited,
-                    COALESCE((SELECT SUM(posts_read) FROM user_visits AS uv2 WHERE uv2.user_id = u.id AND uv2.visited_at > :since), 0) posts_read,
-                    SUM(CASE WHEN t2.id IS NOT NULL AND ua.action_type = :new_topic_type THEN 1 ELSE 0 END) topic_count,
-                    SUM(CASE WHEN p.id IS NOT NULL AND t.id IS NOT NULL AND ua.action_type = :reply_type THEN 1 ELSE 0 END) post_count
-                  FROM users AS u
-                  LEFT OUTER JOIN user_actions AS ua ON ua.user_id = u.id AND COALESCE(ua.created_at, :since) > :since
-                  LEFT OUTER JOIN posts AS p ON ua.target_post_id = p.id AND p.deleted_at IS NULL AND p.post_type = :regular_post_type AND NOT p.hidden
-                  LEFT OUTER JOIN topics AS t ON p.topic_id = t.id AND t.archetype = 'regular' AND t.deleted_at IS NULL AND t.visible
-                  LEFT OUTER JOIN topics AS t2 ON t2.id = ua.target_topic_id AND t2.archetype = 'regular' AND t2.deleted_at IS NULL AND t2.visible
-                  LEFT OUTER JOIN categories AS c ON t.category_id = c.id
-                  WHERE u.active
-                    AND u.silenced_till IS NULL
-                    AND u.id > 0
-                  GROUP BY u.id)
-      UPDATE directory_items di SET
-               likes_received = x.likes_received,
-               likes_given = x.likes_given,
-               topics_entered = x.topics_entered,
-               days_visited = x.days_visited,
-               posts_read = x.posts_read,
-               topic_count = x.topic_count,
-               post_count = x.post_count
-      FROM x
+      DB.exec(<<~SQL, query_args)
```

Comment:
I know this query is a bit harder to follow now, but it is now _much_ easier to debug when we have the SELECT/CTE part separate from the UPDATE.